### ISSUE TITLE
Latest release - 20201214162737 - No longer runs on Windows

### ISSUE THREAD

**Issue by Reelix** [rapid7/metasploit-framework#14512]
1.) Install `metasploit-framework 6.0.22+20201214162737-1rapid7` on Windows 10 20H2
2.) Attempt to run `msfconsole.bat`
3.) Note the following error

```
R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34: warning: Win32API is deprecated after Ruby 1.9.1; use fiddle directly instead
Traceback (most recent call last):
        117: from R:/metasploit-framework/bin/../embedded/framework/msfconsole:23:in `<main>'
        116: from R:/metasploit-framework/embedded/framework/lib/metasploit/framework/command/base.rb:82:in `start'
        115: from R:/metasploit-framework/embedded/framework/lib/metasploit/framework/command/console.rb:48:in `start'
        114: from R:/metasploit-framework/embedded/framework/lib/metasploit/framework/command/console.rb:62:in `driver'
        113: from R:/metasploit-framework/embedded/framework/lib/metasploit/framework/command/console.rb:62:in `new'
        112: from R:/metasploit-framework/embedded/framework/lib/msf/ui/console/driver.rb:164:in `initialize'
        111: from R:/metasploit-framework/embedded/framework/lib/msf/base/simple/framework/module_paths.rb:50:in `init_module_paths'
        110: from R:/metasploit-framework/embedded/framework/lib/msf/base/simple/framework/module_paths.rb:50:in `each'
        109: from R:/metasploit-framework/embedded/framework/lib/msf/base/simple/framework/module_paths.rb:51:in `block in init_module_paths'
        108: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/module_paths.rb:40:in `add_module_path'
        107: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/module_paths.rb:40:in `each'
        106: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/module_paths.rb:41:in `block in add_module_path'
        105: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/loading.rb:130:in `load_modules'
        104: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/loading.rb:130:in `each'
        103: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/loading.rb:132:in `block in load_modules'
        102: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/base.rb:243:in `load_modules'
        101: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/directory.rb:27:in `each_module_reference_name'
        100: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/directory.rb:27:in `foreach'
         99: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/directory.rb:37:in `block in each_module_reference_name'
         98: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rex-core-0.1.14/lib/rex/file.rb:132:in `find'
         97: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rex-core-0.1.14/lib/rex/file.rb:132:in `catch'
         96: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/rex-core-0.1.14/lib/rex/file.rb:133:in `block in find'
         95: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/directory.rb:46:in `block (2 levels) in each_module_reference_name'
         94: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/base.rb:244:in `block in load_modules'
         93: from R:/metasploit-framework/embedded/framework/lib/msf/core/modules/loader/base.rb:189:in `load_module'
         92: from R:/metasploit-framework/embedded/framework/lib/msf/core/module_manager/loading.rb:70:in `on_module_load'
         91: from R:/metasploit-framework/embedded/framework/lib/msf/core/payload_set.rb:208:in `add_module'
         90: from R:/metasploit-framework/embedded/framework/lib/msf/core/payload_set.rb:208:in `new'
         89: from R:/metasploit-framework/embedded/framework/modules/payloads/singles/cmd/unix/reverse_ssh.rb:27:in `initialize'
         88: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         87: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         86: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         85: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `require'
         84: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `require'
         83: from R:/metasploit-framework/embedded/framework/lib/msf/core/handler/reverse_ssh.rb:2:in `<top (required)>'
         82: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         81: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         80: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         79: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         78: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         77: from R:/metasploit-framework/embedded/framework/lib/rex/proto/ssh.rb:4:in `<top (required)>'
         76: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         75: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         74: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         73: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         72: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         71: from R:/metasploit-framework/embedded/framework/lib/rex/proto/ssh/server.rb:2:in `<top (required)>'
         70: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         69: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         68: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         67: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         66: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         65: from R:/metasploit-framework/embedded/framework/lib/rex/proto/ssh/connection.rb:2:in `<top (required)>'
         64: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         63: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         62: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         61: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         60: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         59: from R:/metasploit-framework/embedded/framework/lib/rex/proto/ssh/hrr_rb_ssh.rb:3:in `<top (required)>'
         58: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         57: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         56: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         55: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         54: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         53: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh.rb:18:in `<top (required)>'
         52: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         51: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         50: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         49: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         48: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         47: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection.rb:7:in `<top (required)>'
         46: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         45: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         44: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         43: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         42: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         41: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/channel.rb:6:in `<top (required)>'
         40: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         39: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         38: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         37: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         36: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         35: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/channel/channel_type.rb:19:in `<top (required)>'
         34: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         33: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         32: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         31: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         30: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         29: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/channel/channel_type/session.rb:60:in `<top (required)>'
         28: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         27: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         26: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         25: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         24: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         23: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/channel/channel_type/session/request_type.rb:23:in `<top (required)>'
         22: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         21: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         20: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         19: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         18: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         17: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/channel/channel_type/session/request_type/pty_req.rb:5:in `<top (required)>'
         16: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
         15: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
         14: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
         13: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         12: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
         11: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/request_handler.rb:20:in `<top (required)>'
         10: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
          9: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
          8: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
          7: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
          6: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
          5: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/hrr_rb_ssh-0.3.0.pre2/lib/hrr_rb_ssh/connection/request_handler/reference_pty_req_request_handler.rb:6:in `<top (required)>'
          4: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `require'
          3: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:257:in `load_dependency'
          2: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/activesupport-5.2.4.4/lib/active_support/dependencies.rb:291:in `block in require'
          1: from R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
R:/metasploit-framework/embedded/lib/ruby/gems/2.7.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require': cannot load such file -- pty (LoadError)

```
4.) Uninstall `metasploit-framework 6.0.22+20201214162737-1rapid7`
5.) Install `metasploit-framework 6.0.21+20201207193259-1rapid7`
6.) Run `msfconsole.bat`
7.) Notice that Metasploit Framework starts as expected.

**Comment by mdf3192078** [comment]
The same situation！

**Comment by mdf3192078** [comment]
I have the same situation。

**Comment by adfoster-r7** [comment]
@Reelix This issue impacted 6.0.22 and should hopefully be fixed in 6.0.23. The new release should be built in a few hours from now.
